lib/contexts/services/ServicesContextProvider.tsx

```diff
@@ -0,0 +1,78 @@
+import {PropsWithChildren, useMemo} from "react";
+import {useDataContext} from "../data/DataContext.ts";
+import {OptionDetail, ServiceInitParams} from "../../models.ts";
+import {ServicesContext} from "./ServicesContext.ts";
+import {isRowMapped} from "../../helpers/getters.ts";
+import {updateRow} from "../../services/updateMappingService.ts";
+
+export const ServicesContextProvider = ({
+                                            callback,
+                                            children
+                                        }: PropsWithChildren<ServiceInitParams>) => {
+    const {
+        datasetMapping,
+        headerIndexes,
+        datasetMappingHeader,
+        setDatasetMapping,
+        setDatasetMappingHeader,
+        suggestions,
+    } = useDataContext();
+
+    const datasetMappingService = useMemo(() => {
```

Comment:
@afonsobspinto why do we need to wrap these functions in a context?

Reply:
It is not mandatory that we have this service wrapped in a context. It is just a way to have easy access to the data that the services work on.
I like that with the current implementation the components don't need to do `useDataContext` to get the data to pass for the services but I'm not opposed to change if you prefer otherwise.